website/src/components/articles/article-sections.tsx

```diff
@@ -14,86 +15,92 @@ export const ArticleSections: FunctionComponent<ArticleSectionsProperties> = ({
   data,
 }) => {
   const dispatch = useDispatch();
+  const [activeHeadingId, setActiveHeadingId] = useState<string>();
+  const [headings, setHeadings] = useState<Heading[]>([])
+  const yScrollPosition = useSelector<State, number>((state) =>
+    state.common.yScrollPosition
+  );
 
   const handleCloseClick = useCallback(() => {
     dispatch(closeAside());
   }, []);
 
   useEffect(() => {
-    const ids = data
-      .tableOfContents!.split(/"|\//)
-      .filter((item) => item.indexOf("#") === 0)
-      .map((item) => {
-        const id = item.substring(1);
-
-        return {
-          id: `toc-${id}`,
-          position: document.getElementById(id)!.offsetTop - 80,
-        };
-      })
+    const result = ((data.tableOfContents.items as TableOfContentItem[]) ?? [])
+      .flatMap(item => ([item, ...item.items ?? []]))
+      .map(item => ({
+        id: item.url,
+        position: document.getElementById(item.url.substring(1))!
+          .offsetTop - 80
+      }))
       .reverse();
-    let currentActiveId: string | undefined;
-
-    const handler = () => {
-      const currentScrollPosition =
-        document.body.scrollTop || document.documentElement.scrollTop;
-      let newActiveId: string | undefined;
-
-      for (let i = 0; i < ids.length; i++) {
-        if (currentScrollPosition >= ids[i].position) {
-          newActiveId = ids[i].id;
-          break;
-        }
-      }
-
-      if (currentActiveId !== newActiveId) {
-        if (currentActiveId) {
-          document.getElementById(currentActiveId)!.parentElement!.className =
-            "";
-        }
-
-        currentActiveId = newActiveId;
-
-        if (currentActiveId) {
-          document.getElementById(currentActiveId)!.parentElement!.className =
-            "active";
-        }
-      }
-    };
-
-    if (ids.length > 0) {
-      document.addEventListener("scroll", handler);
-    }
-
-    return () => {
-      if (ids.length > 0) {
-        document.removeEventListener("scroll", handler);
-      }
-    };
+
+    setHeadings(result);
   }, [data]);
 
-  return data.tableOfContents!.length > 0 ? (
+  useEffect(() => {
+    setActiveHeadingId(
+      headings.find(id => yScrollPosition >= id.position)?.id
+    );
+  }, [headings, yScrollPosition]);
+
+  const tocItems: TableOfContentItem[] = data.tableOfContents.items ?? [];
+
+  return tocItems.length > 0 ? (
     <Container>
       <Title>In this article</Title>
       <MostProminentSection>
-        <Content
-          onClick={handleCloseClick}
-          dangerouslySetInnerHTML={{
-            __html: data.tableOfContents!.replace(
-              /href=\"(.*?#)(.*?)\"/gi,
-              'id="toc-$2" href="/docs$1$2"'
-            ),
-          }}
-        />
+        <div onClick={handleCloseClick}>
```

Comment:
Is there a reason for the extra div? Why not using the `onClick` directly on `TableOfContent`?

Reply:
`TableOfContent` is not a styled-component but a normal React one. As far as I know you would need to explicitly declare `onClick` as a prop and drill that to the inner components container. That is not worth it in my opinion.